tests/integration/tests/test_gateway.py

```diff
@@ -47,12 +50,26 @@ def test_gateway(session_instance: harness.Instance):
             "kubectl",
             "get",
             "service",
-            "cilium-gateway-my-gateway",
-            "-o=jsonpath='{.spec.ports[?(@.name==\"port-80\")].nodePort}'",
+            "-o=json",
         ],
         capture_output=True,
     )
-    gateway_http_port = p.stdout.decode().replace("'", "")
+
+    services = json.loads(p.stdout.decode())
+
+    ingress_services = [
+        svc for svc in services["items"] if "my-gateway" in svc["metadata"]["name"]
+    ]
```

Comment:
left the same comment but i don't it has been addressed. first of all, this lists `ingress_services` but the tes is about gateway.

secondly, this check is not accurate, as it's quite easy to get false positives by other resources that exist on the cluster. this would be very hard to track down, and result in bugs finding their way in the code without being found.

We know what is the resource that we just deployed, please check directly for that resource